src/Type/Constant/ConstantArrayType.php

```diff
@@ -532,6 +533,23 @@ public function unsetOffset(Type $offsetType): Type
 		return new ArrayType($this->getKeyType(), $this->getItemType());
 	}
 
+	public function getOffsetType(Type $offsetValueType): Type
```

Comment:
this whole thing is mostly incorrect and weird. and could be the reason for that CI slevomat error that I'm trying to fix right now..